File: Php/app/Homegame/Add/HomegameAddModel.php
<?php
namespace app\Homegame\Add{

	use entities\CurrencySettings;
	use entities\Homegame;
	use Domain\Classes\User;
	use core\PageModel;
	use DateTimeZone;
	use core\FormFields\CurrencyLayoutFieldModel;
	use core\FormFields\TimezoneFieldModel;
	use core\Globalization;

	class HomegameAddModel extends PageModel {

		public $displayName;
		public $description;
		public $currencySymbol;
		public $currencyLayoutSelectModel;
		public $timezoneSelectModel;

		public function __construct(User $user, Homegame $homegame = null){
			parent::__construct($user);
			if($homegame != null){
				$this->displayName = $homegame->getDisplayName();
				$this->description = $homegame->getDescription();
				$this->setTimezoneAndCurrency($homegame->getTimezone(), $homegame->getCurrency());
			} else {
				$this->setTimezoneAndCurrency(Homegame::getDefaultTimezone(), Homegame::getDefaultCurrency());
			}
		}

		private function setTimezoneAndCurrency(DateTimeZone $timezone, CurrencySettings $currency){
			$this->timezoneSelectModel = $this->getTimezoneSelectModel($timezone);
			$this->currencySymbol = $currency->getSymbol();
			$this->currencyLayoutSelectModel = $this->getCurrencyLayoutSelectModel($currency->getLayout());
		}

		private function getTimezoneSelectModel(DateTimeZone $timezone){
			$timezoneNames = Globalization::getTimezoneNames();
			return new TimezoneFieldModel('timezone', 'timezone', $timezone->getName(), $timezoneNames, 'Select Timezone');
		}

		private function getCurrencyLayoutSelectModel($layout){
			return new CurrencyLayoutFieldModel('currencylayout', 'currencylayout', $layout);
		}

	}

}

File: Web/Models/HomegameModels/Add/HomegameAddModel.cs
using System;
using Core.Classes;
using Infrastructure.System;
using Web.Models.FormModels;
using Web.Models.PageBaseModels;

namespace Web.Models.HomegameModels.Add{

	public class HomegameAddModel : PageModel {

	    public string DisplayName { get; set; }
	    public string Description { get; set; }
	    public string CurrencySymbol { get; set; }
	    public CurrencyLayoutFieldModel CurrencyLayoutSelectModel { get; set; }
	    public TimezoneFieldModel TimezoneSelectModel { get; set; }

		public HomegameAddModel(User user, Homegame homegame = null)
            : base (user)
        {
			if(homegame != null){
				DisplayName = homegame.DisplayName;
				Description = homegame.Description;
				SetTimezoneAndCurrency(homegame.Timezone, homegame.Currency);
			} else {
				SetTimezoneAndCurrency(Homegame.DefaultTimezone, Homegame.DefaultCurrency);
			}
		}

		private void SetTimezoneAndCurrency(TimeZoneInfo timezone, CurrencySettings currency){
			TimezoneSelectModel = GetTimezoneSelectModel(timezone);
			CurrencySymbol = currency.Symbol;
			CurrencyLayoutSelectModel = GetCurrencyLayoutSelectModel(currency.Layout);
		}

		private TimezoneFieldModel GetTimezoneSelectModel(TimeZoneInfo timezone){
			var timezones = Globalization.GetTimezones();
			return new TimezoneFieldModel("timezone", "timezone", timezone.Id, timezones, "Select Timezone");
		}

		private CurrencyLayoutFieldModel GetCurrencyLayoutSelectModel(string layout){
			return new CurrencyLayoutFieldModel("currencylayout", "currencylayout", layout);
		}

        public override string BrowserTitle
        {
            get
            {
                return "Create Homegame";
            }
        }

	}

}